src/routes/download.rs

```diff
@@ -1,14 +1,60 @@
 use std::path::PathBuf;
 
+use crate::filename::FileName;
 use crate::RAVEN_REPOSITORY_PATH;
+
 use rocket::response::NamedFile;
+use rocket::response::Responder;
+use rocket::{response, Request, Response};
+
+struct DownloadFile(NamedFile, FileName, FileName, FileName);
+
+impl<'r> Responder<'r> for DownloadFile {
+    fn respond_to(self, req: &Request) -> response::Result<'r> {
+        Response::build_from(self.0.respond_to(req)?)
+            .raw_header(
```

Comment:
Use `format!` instead of using the `+` and having fun chaining functions like `into()` :p 

Eg:
```rust
format!("attachment; filename=\"{}-{}-{}.tar.gz\"", self.1, self.2, self.3);
```

(I told you, if something you write lacks elegance, there is probably a better way. Don't hesitate to ask!)